src/pages/patientView/pathwayMapper/PatientViewPathwayMapper.tsx

```diff
@@ -0,0 +1,291 @@
+                                    this.addGenomicDataHandler
+                                }
+                                tableComponent={this.renderTable}
+                                validGenes={this.validGenes}
+                                patientView={true}
+                                //message banner patch will be removed
+                                messageBanner={this.renderBanner}
+                            />
+                            <ToastContainer
+                                closeButton={<i className="fa fa-times" />}
+                            />
+                        </React.Fragment>
+                    </Row>
+                </div>
+            </div>
+        );
+    }
+
+    @computed get storeForAllData(): PatientViewPageStore | undefined {
+        let patientStore: PatientViewPageStore | undefined;
+        if (this.urlWrapperForAllGenes) {
+            return patientStore;
+        } else {
+            return undefined;
+        }
+    }
+
+    /**
+     * Here we clone the "query" field of the main store's URL Wrapper and enhance the cloned query
+     * with additional non-query genes. This new query object is used to fake a new URL Wrapper instance
+     * which is required to initialize a separate ResultsViewStore for the non-query genes.
+     */
+    @computed get urlWrapperForAllGenes(): PatientViewUrlWrapper | undefined {
+        let urlWrapper: PatientViewUrlWrapper | undefined;
+        if (
+            this.validNonQueryGenes.isComplete &&
+            this.validNonQueryGenes.result.length > 0
+        ) {
+            // fake the URL wrapper, we only need the query parameters with additional genes
+            const query: { [key: string]: any } = _.cloneDeep(
+                this.props.urlWrapper.query
+            );
+            query.gene_list = this.validNonQueryGenes.result.join(' ');
+
+            // we don't need a proper URL Wrapper here, just assign an object with a valid query field
+            urlWrapper = { query } as PatientViewUrlWrapper;
+        }
+
+        return urlWrapper;
+    }
+    @computed get validGenes() {
+        if (this.validNonQueryGenes.isComplete) {
+            // Valid genes are accumulated.
+            this.validNonQueryGenes.result.forEach(gene => {
+                this.accumulatedValidGenes[gene] = true;
+            });
+        }
+        return this.accumulatedValidGenes;
+    }
+    /**
+     * addGenomicData function is implemented in PathwayMapper component and overlays
+     * alteration data onto genes. Through this function callback, the function implemented
+     * in PathwayMapper is copied here.
+     */
+    @autobind
+    @action
+    private addGenomicDataHandler(
+        addGenomicData: (alterationData: ICBioData[]) => void
+    ) {
+        this.addGenomicData = addGenomicData;
+    }
+
+    @autobind
+    @action
+    private handlePathwayChange(pathwayGenes: string[]) {
```

Comment:
This also seems redundant for patient view. We do not need to store this information.